lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,240 @@
+        if repo_enabled_re:
+            repo_enabled = repo_enabled_re.group(1)
+
+            repo = {"id": repo_id, "name": repo_name, "url": repo_url, "enabled": True if repo_enabled == '1' else False}
+            repo_result.append(repo)
+    return repo_result
+
+
+def repository_list(module, list_parameter):
+    # Get rhsm repository list and format it for the user
+    repo = get_repository_list(module, list_parameter)
+    module.exit_json(changed=False, repositories=repo)
+
+
+def repository_modify(module, state, name):
+    current_repo_list = get_repository_list(module, 'all')
+    name = set(name)
+
+    matched_existing_repo = {}
+    for repoid in name:
+        matched_existing_repo[repoid] = []
+        for repo in current_repo_list:
+            if fnmatch(repo['id'], repoid):
+                matched_existing_repo[repoid].append(repo)
+
+    changed = False
+    results = []
+    rhsm_arguments = ['repos']
+
+    for repoid in matched_existing_repo:
+        if len(matched_existing_repo[repoid]) == 0:
+            results.append("%s is not a valid repository ID." % repoid)
+            module.fail_json(results=results, msg="%s is not a valid repository ID." % repoid)
+        for repo in matched_existing_repo[repoid]:
+            if state == 'disabled':
+                if repo['enabled']:
+                    changed = True
+                if module.check_mode:
+                    results.append("Repository '%s' is disabled for this system." % repo['id'])
+                else:
+                    rhsm_arguments += ['--disable', repoid]
+            elif state == 'enabled':
+                if not repo['enabled']:
+                    changed = True
+                if module.check_mode:
+                    results.append("Repository '%s' is enabled for this system." % repo['id'])
+                else:
+                    rhsm_arguments += ['--enable', repoid]
+
+    if module.check_mode:
+        module.exit_json(results=results, changed=changed)
+    rc, out, err = run_subscription_manager(module, rhsm_arguments)
+    results = out.split('\n')
+    module.exit_json(results=results)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default=None,
+                       choices=['list', 'enabled', 'disabled'],
+                       required=True),
+            name=dict(type="list")
+        ),
+        supports_check_mode=True
+    )
+    state = module.params['state']
+    name = module.params['name']
+
+    if state in ['enabled', 'disabled'] and name is None:
+        err_msg = "argument name is required when state is enabled or disabled"
+        module.fail_json(msg=err_msg)
+
+    if state == 'list':
+        if isinstance(name, list) and (len(name) > 1 or name[0] not in ['all', 'enabled', 'disabled']):
+            err_msg = 'when state is list, the allowed value for the option value are: all, enabled, disabled'
```

Comment:
Thanks for the review.

I have applied the suggestion reported in the your previous five comment in my last commit just pushed.
As standard to make conform capitalization and punctuation in every error message, I used yum module as example:
- Capital letter for the first characther of each message
- No dot at the end of the message